adapters/visx/visx.go

```diff
@@ -14,11 +14,38 @@ type VisxAdapter struct {
 	endpoint string
 }
 
+type visxBid struct {
+	ImpID   string   `json:"impid"`
+	Price   float64  `json:"price"`
+	UID     int      `json:"auid"`
+	AdM     string   `json:"adm,omitempty"`
+	ADomain []string `json:"adomain,omitempty"`
+	DealID  string   `json:"dealid,omitempty"`
+	W       uint64   `json:"w,omitempty"`
+	H       uint64   `json:"h,omitempty"`
+}
+
+type visxSeatBid struct {
+	Bid  []visxBid `json:"bid"`
+	Seat string    `json:"seat,omitempty"`
+}
+
+type visxResponse struct {
+	SeatBid []visxSeatBid `json:"seatbid,omitempty"`
+}
+
 // MakeRequests makes the HTTP requests which should be made to fetch bids.
 func (a *VisxAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
 	var errors = make([]error, 0)
 
-	reqJSON, err := json.Marshal(request)
+	// copy the request, because we are going to mutate it
+	requestCopy := *request
+	if len(requestCopy.Cur) == 0 {
+		requestCopy.Cur = make([]string, 1)
```

Comment:
This is fine. An alternative way of writing this is the one liner: `requestCopy.Cur = []string{"USD"}`

Reply:
Done.